packages/e2e/lib/e2e.dart

```diff
@@ -184,4 +241,88 @@ class E2EWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding {
     );
     results[description] ??= 'success';
   }
+
+  /// Handle commands from WebDriver tests.
+  ///
+  /// WebDriver provides a set methods (ex: screenshot, resize, refresh) which
+  /// are useful for writing end-to-end tests on multiple browsers.
+  ///
+  /// e2e tests are sending [TestStatus] messages to state that they are ready
+  /// to run a WebDriver action ([TestStatus.wait_on_webdriver_command]) or
+  /// they are finished a web driver action
+  /// ([TestStatus.webdriver_command_complete])
+  ///
+  /// The tests running on the browser sends request these WebDriver operations
+  /// from e2e tests via replying with [WebDriverAction]s.
+  Future<Map<String, dynamic>> _callbackForWebDriver(
+      Map<String, String> params) async {
+    final String command = params['command'];
+    // Test status is added as an exta message.
+    final String extraMessage = params['message'] ?? '';
+    Map<String, String> response;
+    switch (command) {
```

Comment:
IOW, can we avoid having this switch statement that will continue to grow, with nested `if`s underneath that will get harder to reason about?

Reply:
I refactored the methods and separated the part for request_data into 2 methods. I hope it looks more readable now. PHAL.